java/deepgramLanguageDetection/Index.java

```diff
@@ -0,0 +1,172 @@
+import java.util.Collections;
+import java.util.stream.Collectors;
+import java.util.*;
+import java.util.stream.*;
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+
+import com.google.gson.Gson;
+import org.apache.commons.validator.routines.UrlValidator;
+final Gson gson=new Gson();
+
+public RuntimeResponse main(RuntimeRequest req,RuntimeResponse res)throws Exception{
+        // Validate that values present in the request are not empty (payload, variables)
+        RuntimeResponse errorResponse=checkEmptyPayloadAndVariables(req,res);
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        // Validate the requested payload (URL)
+        String payloadString=req.getPayload().toString();
+        Map<String, Object> payload=gson.fromJson(payloadString,Map.class);
+
+        errorResponse=validatePayload(payload,res);
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        // Get file url from payload
+        String fileurl=payload.get("fileUrl").toString();
+
+        // Name API key is not empty
+        String apiKeyVariable="DEEPGRAM_SECRET_KEY";
+
+        errorResponse=checkEmptyAPIKey(req,res,apiKeyVariable);
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        String apiKey=req.getVariables().get(apiKeyVariable).toString();
+
+        String deepgramData="";
+        Map<String, Object> responseData=new HashMap<>();
+
+        try{
+            deepgramData=detectLanguage(fileurl,apiKey);
+        }catch(Exception e){
+            responseData.put("success",false);
+            responseData.put("message","Something went wrong while generating the summary, please check with the developers. Error: "+e.getMessage());//TODO
+        return res.json(responseData);
+        }
+        responseData.put("success",true);
```

Comment:
Can we add a check for the status code within the Deepgram API's request and send a success/failure response accordingly?

In its current state, the function might return a successful response even if Deepgram's API returns a Bad Request or Unauthorized error.

Reply:
400s error gets handled in try catch,
![Screenshot from 2022-10-29 22-33-43](https://user-images.githubusercontent.com/64471630/198844166-88e0c1bb-96ac-4321-be77-0a2cf2b2e38c.png)